tests/lib/ocpp/v201/test_smart_charging_handler.cpp

```diff
@@ -0,0 +1,276 @@
+            chargingSchedules,
+            {}, // transactionId
+            recurrencyKind,
+            {}, // validFrom
+            {}  // validTo
+        };
+    }
+
+    DeviceModel createDeviceModel() {
+        std::unique_ptr<DeviceModelStorageMock> storage_mock =
+            std::make_unique<testing::NiceMock<DeviceModelStorageMock>>();
+        ON_CALL(*storage_mock, get_device_model).WillByDefault(testing::Return(DeviceModelMap()));
+        return DeviceModel(std::move(storage_mock));
+    }
+
+    void createEvseWithId(int id) {
+        testing::MockFunction<void(const MeterValue& meter_value, const Transaction& transaction, const int32_t seq_no,
+                                   const std::optional<int32_t> reservation_id)>
+            transaction_meter_value_req_mock;
+        testing::MockFunction<void()> pause_charging_callback_mock;
+        auto e1 = std::make_unique<Evse>(
+            id, 1, device_model, database_handler, std::make_shared<ComponentStateManagerMock>(),
+            transaction_meter_value_req_mock.AsStdFunction(), pause_charging_callback_mock.AsStdFunction());
+        evses[id] = std::move(e1);
+    }
+
+    SmartChargingHandler createSmartChargingHandler() {
+        return SmartChargingHandler(database_handler);
+    }
+
+    std::string uuid() {
+        std::stringstream s;
+        s << uuid_generator();
+        return s.str();
+    }
+
+    void open_evse_transaction(int evseId, std::string transactionId) {
+        auto connector_id = 1;
+        auto meter_start = MeterValue();
+        auto id_token = IdToken();
+        auto date_time = ocpp::DateTime("2024-01-17T17:00:00");
+        evses[evseId]->open_transaction(
+            transactionId, connector_id, date_time, meter_start, id_token, {}, {},
+            std::chrono::seconds(static_cast<int64_t>(1)), std::chrono::seconds(static_cast<int64_t>(1)),
+            std::chrono::seconds(static_cast<int64_t>(1)), std::chrono::seconds(static_cast<int64_t>(1)));
+    }
+    // Default values used within the tests
+    std::map<int32_t, std::unique_ptr<Evse>> evses;
+    std::shared_ptr<DatabaseHandler> database_handler;
+
+    const int evseId = 1;
+    bool ignore_no_transaction = true;
+    const int profile_max_stack_level = 1;
+    const int max_charging_profiles_installed = 1;
+    const int charging_schedule_max_periods = 1;
+    DeviceModel device_model = createDeviceModel();
+    SmartChargingHandler handler = createSmartChargingHandler();
+    boost::uuids::random_generator uuid_generator = boost::uuids::random_generator();
+};
+
+TEST_F(ChargepointTestFixtureV201, K01FR03_IfTxProfileIsMissingTransactionId_ThenProfileIsInvalid) {
+    createEvseWithId(evseId);
+    auto profile = createTxProfileWithMissingTransactionId(createChargeSchedule(ChargingRateUnitEnum::A));
+    auto sut = handler.validate_tx_profile(profile, *evses[evseId]);
+
+    ASSERT_THAT(sut, testing::Eq(ProfileValidationResultEnum::TxProfileMissingTransactionId));
```

Comment:
Wouln't it be better to use `EXPECT_EQ` here so testing can continue if one of these fail? I think that's also preferred from google test

Reply:
Changed to `EXPECT_THAT`

Reply:
Any reason for preferring `EXPECT_THAT` above `EXPECT_EQ`?

Reply:
EXPECT_EQ() takes two parameters, but it isn't obvious which one is the expected and which one is the actual.

EXPECT_THAT() reads more naturally, and it is obvious which one is the expected value and which one is the actual value.